neqo-transport/src/recovery.rs

```diff
@@ -543,6 +560,53 @@ impl LossRecovery {
             .min_by_key(|&(_, time)| time)
             .map(|(spc, val)| (spc, val + self.loss_delay()))
     }
+
+    /// Find when the last sent ack eliciting packete.
+    pub fn get_min_pto(&self) -> Option<Instant> {
+        // TODO ignore PNSpace::Application until handshake is done -> a server side problem.
+        PNSpace::iter()
+            .filter_map(|spc| {
+                if self.spaces[*spc]
+                    .sent_packets
+                    .values()
+                    .any(|sp| sp.ack_eliciting)
+                {
+                    // None is ordered less than Some(_). Bad. Filter them out.
+                    if let Some(time) = self.spaces[*spc].time_of_last_sent_ack_eliciting_packet {
+                        Some(time + self.pto_timeout(*spc))
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                }
+            })
+            .min_by_key(|&time| time)
+    }
+
+    pub fn get_min_pto_pn_space(&self, now: Instant) -> Option<PNSpace> {
+        PNSpace::iter()
+            .filter_map(|spc| {
+                if self.spaces[*spc]
+                    .sent_packets
+                    .values()
+                    .any(|sp| sp.ack_eliciting)
```

Comment:
Can we avoid doing this iteration on every PTO by recording this information on the object when we send a packet?  We could clear it when we process acknowledgments.

Which do you think would be easier to manage?

I you keep it this way around, I would prefer to add a function on LossRecoverySpace that performs this check.  `ack_eliciting_outstanding() -> bool` would help a lot.

Reply:
this is call only if PTO timer expires (or if app calls before PTO timer expires and pto time is set - if app is doing it right it should not happe). I do not think wee will do PTO a lot so this function is not a concern, but I think there is one other place that is problematic (in the code that this PR have not touched). I will look into fixing it.